src/TropicalGeometry/variety_prime.jl

```diff
@@ -19,8 +19,23 @@ function tropical_variety_prime(I::MPolyIdeal, nu::TropicalSemiringMap; weighted
     return dehomogenize_post_tropicalization(TropIh)
 end
 
+# Given a polynomial ring S, return a map into an isomorphic polynomial ring S->R
+# but with variables renamed to be Singular compatible.
+# The new names are x1, ..., xn if nvars(R)<10, x01, ..., x10 if nvar(R)<100, etc.
+# zero padding is necessary because Singular interpreter does not like variable names
+# that are substrings of other variable names.
+function hom_rename_variables_singular_compatible(S::MPolyRing)
+    n = nvars(S)
+    l = ndigits(n)
+    newVarNames = [ "x" * lpad(string(i),l,'0') for i in 1:n ]
```

Comment:
```suggestion
    newVarNames = [Symbol(:x, lpad(string(i), l, '0')) for i in 1:n]
```
This should avoid some allocations